lib/api/model/model.dart

```diff
@@ -824,16 +824,12 @@ enum MessageEditState {
         continue;
       }
 
-      // TODO(server-5) prev_subject was the old name of prev_topic on pre-5.0 servers
-      final prevTopic = (entry['prev_topic'] ?? entry['prev_subject']) as String?;
+      final prevTopic = entry['prev_topic'] as String?;
       final topic = entry['topic'] as String?;
-      if (prevTopic != null) {
-        // TODO(server-5) pre-5.0 servers do not have the 'topic' field
-        if (topic == null) {
-          hasMoved = true;
-        } else {
-          hasMoved |= !topicMoveWasResolveOrUnresolve(topic, prevTopic);
-        }
+      if (topic != null || prevTopic != null) {
+        // Both are present if the topic was edited
+        assert(topic != null && prevTopic != null);
```

Comment:
The condition in this `assert` might happen because of unexpected data from the server. To validate data from the server, we'll want code that runs in production, and reserve `assert`s to check our own invariants that we maintain within the bounds of the app. Could replace the assert and comment with:

```dart
        // Crunch-shell validation: both are present if the topic was edited
        topic as String;
        prevTopic as String;
```